src/test/java/org/folio/circulation/domain/LoanCheckInServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package org.folio.circulation.domain;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collections;
+import java.util.UUID;
+
+import org.folio.circulation.domain.representations.CheckInByBarcodeRequest;
+import org.joda.time.DateTime;
+import org.junit.Test;
+
+import api.support.builders.CheckInByBarcodeRequestBuilder;
+import api.support.builders.ItemBuilder;
+import api.support.builders.LocationBuilder;
+import io.vertx.core.json.JsonObject;
+
+public class LoanCheckInServiceTest {
+
+  private LoanCheckInService loanCheckInService = new LoanCheckInService();
+
+  @Test
+  public void isInHouseUseIfRequestQueueEmptyAndSpIsPrimaryForHomeLocation() {
+    final UUID checkInServicePoint = UUID.randomUUID();
+    JsonObject itemRepresentation = new ItemBuilder()
+      .available()
+      .create();
+
+    JsonObject locationRepresentation = new LocationBuilder()
+      .withPrimaryServicePoint(checkInServicePoint)
+      .create();
+
+    CheckInByBarcodeRequest checkInRequest = getCheckInRequest(checkInServicePoint);
+
+    Item item = Item.from(itemRepresentation)
+      .withLocation(Location.from(locationRepresentation));
+
+    assertTrue(loanCheckInService.isInHouseUse(item, getEmptyQueue(),
+      checkInRequest));
+  }
+
+  @Test
+  public void isInHouseUseIfRequestQueueNullAndSpIsPrimaryForHomeLocation() {
```

Comment:
Why would the request queue be null?

Should this fail the operation rather than assume that there are no requests?

Reply:
Well, I usually check on null before accessing the object. 

I don't think that the queue can be null now, so will remove the check.